Services/TMS/CourseActions/ToCourse.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/* Copyright (c) 2018 Stefan Hecken <stefan.hecken@concepts-and-training.de> */
+
+use ILIAS\TMS;
+
+/**
+ * This is an information about a course action, noteworthy for a user in some context.
+ */
+class ToCourse extends TMS\CourseActionImpl
+{
+	/**
+	 * @inheritdoc
+	 */
+	public function isAllowedFor($usr_id)
+	{
+		$course = $this->entity->object();
+		return $this->hasAccess($course->getRefId());
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getLink(\ilCtrl $ctrl, $usr_id)
+	{
```

Comment:
weder $crtrl noch $usr_id werden verwendet

Reply:
Ja stimmt. Die Funktion wird aber vom Interface so bestimmt. Es gibt ein paar Aktionen wo sie benötigt wird. In dem Fall leider nicht. Die ctrl darf laut ILIAS Regeln nur in GUI Klassen verwendet werden. Deshalb hab ich die hier so rein geschubst.